### PR TITLE
feat(apigateway): auto-create RequestValidator from options to addMethod()

### LINKED ISSUE #6193: RequestValidator shouldn't require (or even expect) a restApi property.
Specifying a requestModel in the MethodOptions of an AddMethod() operations should create the RequestValidator for me. Currently I have to also manually construct a RequestValidator.

Currently I have to build something like this:
```
    languageResource.addMethod("POST", new apigw.Integration({
      type: apigw.IntegrationType.AWS,
      uri: `arn:aws:apigateway:${cdk.Aws.REGION}:events:path//`,
      integrationHttpMethod: "POST",
      options: options,
    }),
    {
      methodResponses: [{
          statusCode: "200"
        }],
      requestModels: {"application/json": model.getModel(this, myRestAPI) },
      requestValidator: new apigw.RequestValidator(this, "myValidator", {
        restApi: myRestAPI,
        validateRequestBody: true
      })
    })
```

The validator is being added to a resource, which is inherently attached to a RestAPI. Asking users to specify it again adds no value and we should just add it for them.

### PR DESCRIPTION
**Pull request by nirvana124**
## Commit Message
feat(apigateway): auto-create RequestValidator from options to addMethod() (#6780)

closes #6193
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 1]
```diff
@@ -73,6 +73,7 @@ export interface MethodOptions {
 
   /**
    * The ID of the associated request validator.
+   * @deprecated Use `reqValidator` to pass validator options
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]: Actually, let's not deprecate this yet. Let's keep this around. It'll support cases when a `RequestValidator` is imported, or a user wants to use the same request validator in different methods.

I'm not sure yet that these are actual use cases, that people want.

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 2]
```diff
@@ -73,6 +73,7 @@ export interface MethodOptions {
 
   /**
    * The ID of the associated request validator.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]:
```suggestion
   * The ID of the associated request validator.
   * Only one of `requestValidator` or `requestValidatorOptions` must be specified.
```

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 3]
```diff
@@ -83,6 +84,12 @@ export interface MethodOptions {
    * @default - no authorization scopes
    */
   readonly authorizationScopes?: string[]
+
+  /**
+   * Request validator options to create new validator
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]:
```suggestion
   * Request validator options to create new validator
   * Only one of `requestValidator` or `requestValidatorOptions` must be specified.
```

File: packages/@aws-cdk/aws-apigateway/lib/method.ts [hunk 4]
```diff
@@ -302,6 +309,16 @@ export class Method extends Resource {
 
     return models;
   }
+
+  private requestValidatorId(options: MethodOptions): string | undefined {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/method.ts]: Fail here if both are specified.

File: packages/@aws-cdk/aws-apigateway/README.md [hunk 5]
```diff
@@ -307,8 +307,15 @@ resource.addMethod('GET', integration, {
...
-  // We need to set the validator for ensuring they are passed
+  // To specify a request validator we can set either `requestValidator` or `requestValidatorOptions`
+  // We can set the validator for ensuring they are passed
   requestValidator: validator,
+  // Or we can set request validator options like below
+  requestValidatorOptions: {
+    requestValidatorName: 'test-validator',
+    validateRequestBody: true,
+    validateRequestParameters: false
+  }
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/README.md:318]: This gets confusing. Instead, keep the `requestValidatorOptions` here and drop `requestValidator`.

Add a paragraph after this code block stating something like - "Specifying `requestValidatorOptions` automatically creates the `RequestValidator` construct with the given options. However, if you have your `RequestValidator` already initialized or imported, use the `requestValidator` option instead."

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 6]
```diff
@@ -802,6 +802,84 @@ export = {
...
+          requestValidatorName: 'test-validator',
+          validateRequestBody: true,
+          validateRequestParameters: false
+        }
+      }
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::ApiGateway::RequestValidator', {
+      RestApiId: { Ref: stack.getLogicalId(api.node.findChild('Resource') as cdk.CfnElement) },
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts:828]: This should work, no?

```suggestion
      RestApiId: stack.resolve(api.restApiId),
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 7]
```diff
@@ -802,6 +802,84 @@ export = {
...
+      ValidateRequestBody: true,
+      ValidateRequestParameters: false,
+      Name: 'test-validator'
+    }));
+
+    test.done();
+  },
+
+  'method do not have a request validator'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts:837]:
```suggestion
  'method does not have a request validator'(test: Test) {
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 8]
```diff
@@ -802,6 +802,84 @@ export = {
...
+
+    // THEN
+    expect(stack).to(haveResource('AWS::ApiGateway::Method', {
+      RequestValidatorId: ABSENT
+    }));
+
+    test.done();
+  },
+
+  'method do not support both request validator and request validator options'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts:856]:
```suggestion
  'method does not support both request validator and request validator options'(test: Test) {
```

File: packages/@aws-cdk/aws-apigateway/test/test.method.ts [hunk 9]
```diff
@@ -802,6 +802,84 @@ export = {
...
+          validateRequestBody: true,
+          validateRequestParameters: false
+        },
+        requestValidator: validator
+      }
+    };
+
+    // THEN
+    test.throws(() => new apigw.Method(stack, 'method', methodProps),
+    /Only one of 'requestValidator' or 'requestValidatorOptions' must be specified./);
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/test/test.method.ts:881]:
```suggestion
    test.throws(() => new apigw.Method(stack, 'method', methodProps),
      /Only one of 'requestValidator' or 'requestValidatorOptions' must be specified./);
```

File: packages/@aws-cdk/aws-apigateway/README.md (hunk 5 above)
Comment (nirvana124) [reply packages/@aws-cdk/aws-apigateway/README.md]: Done

**Comment by nirvana124** [comment]
@nija-at Sorry i merged the master into my branch after your approval. Bot dismissed your approval.
Can you please approve it again.

**Pull request merged**